kie-soup-dataset/kie-soup-dataset-sql/src/test/java/org/dashbuilder/dataprovider/sql/SQLDataSetProviderTest.java

```diff
@@ -0,0 +1,105 @@
+ */
+package org.dashbuilder.dataprovider.sql;
+
+import static org.hamcrest.CoreMatchers.endsWith;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.dashbuilder.dataprovider.sql.dialect.Dialect;
+import org.dashbuilder.dataprovider.sql.model.Column;
+import org.dashbuilder.dataset.ColumnType;
+import org.dashbuilder.dataset.DataSetMetadata;
+import org.dashbuilder.dataset.def.DataSetDefFactory;
+import org.dashbuilder.dataset.def.SQLDataSetDef;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({JDBCUtils.class})
+public class SQLDataSetProviderTest {
+
+    @Mock
+    Connection connection;
+
+    @Mock
+    Dialect dialect;
+    
+    @Mock
+    Statement statement;
+    
+    @Spy
+    SQLDataSetProvider dataSetProvider = SQLDataSetProvider.get();
+    
+    @Mock
+    ResultSet resultSet;
+    
+    SQLDataSetProvider sqlDataSetProvider;
+    ResultSetHandler resultSetHandler;
+    SQLDataSetDef dataSetDef = new SQLDataSetDef();
+
+    @Before
+    public void setUp() throws Exception {
+        PowerMockito.mockStatic(JDBCUtils.class);
+        
+        final List<Column> dbColumns = new ArrayList<Column>();
+        dbColumns.add(SQLFactory.column("dbWins1", ColumnType.LABEL, 10));
+        dbColumns.add(SQLFactory.column("dbWins2", ColumnType.LABEL, 10));
+        dbColumns.add(SQLFactory.column("dbWins3", ColumnType.LABEL, 10));
+
+        dataSetProvider.setDataSourceLocator(new DatabaseTestSettings().getDataSourceLocator());
+        resultSetHandler = new ResultSetHandler(resultSet, statement);
+        
+        when(JDBCUtils.dialect(any(Connection.class))).thenReturn(JDBCUtils.H2);
+        when(JDBCUtils.executeQuery(any(Connection.class), any())).thenReturn(resultSetHandler);
+        when(JDBCUtils.getColumns(any(),any())).thenReturn(dbColumns);
+    }
+
+    @Test()
+    public void testGetColumnsWithLimitZero() throws Exception {
+        DataSetMetadata metadata = dataSetProvider.getDataSetMetadata(
+                DataSetDefFactory.newSQLDataSetDef()
+                        .uuid("test")
+                        .estimateSize(false)
+                        .dbTable("test", true)
+                        .buildDef());
+
+        ArgumentCaptor<Connection> arg1 = ArgumentCaptor.forClass(Connection.class);
+        ArgumentCaptor<String> arg2 = ArgumentCaptor.forClass(String.class);
```

Comment:
same, maybe something like
```suggestion
        ArgumentCaptor<String> sql = ArgumentCaptor.forClass(String.class);
```

Reply:
done